### ISSUE TITLE
Gamble Shield item filter

### ISSUE THREAD

**Issue by DomMerlin** [snowgears/shopbugs#386]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Type of bug

Exploit

### Environment

```markdown
- Server Version: 1.19.2 (paper v. 307)
- Shop Version: 1.8.4.2
```

### Server startup log

https://paste.gg/p/anonymous/866fd2a6d838428798aa1fa61e73615b

### Shop config files

https://paste.gg/p/anonymous/5774bb51e4c845409cdbaaed92d9549a

### Bug description

Item filter bug: get only the items you really want to get

### Steps to reproduce

- Pack the "desired" stackable items with a single copy in the inv.
- Fill up the inventory with non stackable items or full stacks
- Hold item lotto chest with right click or click diligently

-> You get the message that there is no space in the inv and nothing is deducted. If you would draw an item from the item lotto, which fits into the inv (because the drawn item "fits" on the stack), you get this item.

### Expected behaviour

If the inventory is full, the plugin should directly give the error message that the inventory is full and should NOT start a randomize.

### Actual behaviour

Players can select what they get from the gamble shop

**Comment by snowgears** [comment]
I am unable to reproduce this bug.

I made a new gamble shop that only has the options for dirt and stone.
![image](https://user-images.githubusercontent.com/4141199/215543016-884a5293-f94e-444d-b1c1-22d2dc4c8ff3.png)

I filled all spaces in my inventory with other items and made sure to put a stack of 1 dirt so that if I got dirt in the gamble output it would be able to fit in my inventory. I am still unable to gamble with the chest. Can you make sure you download the latest file from Spigot? I uploaded a patch fix (still 1.8.4.2 but it contained a bug fix) and deleted the old file

**Comment by snowgears** [comment]
Marking as fixed until I hear otherwise :)
